dist/steps/run_tests.sh

```diff
@@ -7,6 +7,65 @@
 UNITY_PROJECT_PATH="$GITHUB_WORKSPACE/$PROJECT_PATH"
 echo "Using project path \"$UNITY_PROJECT_PATH\"."
 
+#
+# Create an empty project for testing if in package mode
+#
+
+if [ "$PACKAGE_MODE" = "true" ]; then
+  echo "Running tests on a Unity package rather than a Unity project."
+
+  echo ""
+  echo "###########################"
+  echo "#    Package Folder       #"
+  echo "###########################"
+  echo ""
+
+  ls "$UNITY_PROJECT_PATH"
+  echo ""
+
+  PACKAGE_JSON_PATH="$UNITY_PROJECT_PATH/package.json"
+  if [ ! -f "$PACKAGE_JSON_PATH" ]; then
+      echo "Unable to locate package.json from the given package path. Aborting..."
+      exit 1
+  fi
+
+  PACKAGE_NAME=$(cat "$PACKAGE_JSON_PATH" | jq -r ".name")
+
+  if [ -z $PACKAGE_NAME ]; then 
+    echo "Unable to parse package name from package.json. Aborting..."
+    exit 1
+  fi
+
+  echo "Package name found: ${PACKAGE_NAME}"
```

Comment:
Ideally these checks would move to the javascript side. That way 
- we don't have to pull 5GB if we know it's going to fail in the first place
- it will be easier to continue with our efforts to consolidate core logic from test-runner and builder
- we don't need to install jq on runtime or in the images.

Reply:
I agree with moving these checks over to the javascript side - however:

> we don't need to install jq on runtime or in the images.

I don't think this is necessarily true. We still need `jq` for the operation of adding the package name to the generated project's manifest and testables:

```
  PACKAGE_MANIFEST_JSON=$(cat "$PACKAGE_MANIFEST_PATH")
  echo "$PACKAGE_MANIFEST_JSON" | \
    jq \
    --arg packageName "$PACKAGE_NAME" \
    --arg projectPath "$UNITY_PROJECT_PATH" \
    '.dependencies += {"\($packageName)": "file:\($projectPath)"} | . += {testables: ["\($packageName)"]}' \
    > "$PACKAGE_MANIFEST_PATH"

  UNITY_PROJECT_PATH="./TempProject"
```

Granted, we don't really _need_ `jq` per se, but it just saves us from having to manually find and insert the necessary strings into the manifest file. Let me know if you'd prefer that route over using `jq` and I'll work at making that change!

Reply:
I didn't say that we don't need `jq`. The operative word here was meant to be `install during runtime`.

Javascript has npm packages to simplify working with JSON. Though that said javascript natively understands JSON, so you can just [parse](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse) it.

Reply:
Makes sense, sorry for the confusion!

Just to clarify, would I install `jq` in the `Dockerfile`, or would we install it in [the Docker images themselves](https://github.com/game-ci/docker)

Thank you!

Reply:
Indeed, in the docker images themselves.

https://packages.ubuntu.com/bionic/jq 

The size seems to be very small so it's ok to add it to  the base image (`images/ubuntu/base`)